packages/lmi/tests/test_llms.py

```diff
@@ -668,6 +668,60 @@ async def test_empty_tools(self, tools: list | None, model_name: str) -> None:
             assert not result.messages[0].tool_calls
 
 
+class TestReasoning:
+    @pytest.mark.vcr(match_on=[*VCR_DEFAULT_MATCH_ON, "body"])
+    @pytest.mark.asyncio
+    async def test_deepseek_model(self) -> None:
+        llm = LiteLLMModel(
+            name="deepseek/deepseek-reasoner",
+            config={
+                "model_list": [
+                    {
+                        "model_name": "deepseek/deepseek-reasoner",
+                        "litellm_params": {
+                            "model": "deepseek/deepseek-reasoner",
+                            "api_base": "https://api.deepseek.com/v1",
+                        },
+                    }
+                ]
+            },
+        )
+        messages = [
+            Message(
+                role="system",
+                content="Think deeply about the following question and answer it.",
+            ),
+            Message(content="What is the meaning of life?"),
+        ]
+        results = await llm.call(messages)
+        for result in results:
+            assert result.reasoning_content
+
+        outputs: list[str] = []
+        results = await llm.call(messages, callbacks=[outputs.append])
+        for result in results:
+            assert result.reasoning_content
+
+    @pytest.mark.vcr
+    @pytest.mark.asyncio
+    async def test_openrouter_reasoning(self) -> None:
+        llm = LiteLLMModel(name="openrouter/deepseek/deepseek-r1", config={"n": 1})
+        messages = [
+            Message(content="What is the meaning of life?"),
+        ]
+        results = await llm.call(messages)
+        assert isinstance(results, list)
+        assert isinstance(results[0], LLMResult)
+        assert results[0].reasoning_content
+
+        outputs: list[str] = []
+        with pytest.raises(
+            NotImplementedError,
+            match=r"Reasoning with OpenRouter is not supported in streaming mode.*",
+        ):
+            await llm.call(messages, callbacks=[outputs.append])
```

Comment:
Can we remove `callbacks` from here? Or can we use it?

It feels weird to have callbacks that are unused. If trying to test some control flow with this, then maybe add a comment stating that